### PR TITLE
Another refactoring of the consoles

### PR DESCRIPTION
**Pull request by coolo**
This is providing a very basic backend, to be filled

### CODE REVIEW

File: consoles/console.pm [hunk 1]
```diff
@@ -34,6 +34,13 @@ sub screen {
     return;
 }
 
+# convenince function
```
Comment (okurz) [inline consoles/console.pm]: s/convenince/convenience/, let's call it "helper" :-)

File: consoles/localXvnc.pm [hunk 2]
```diff
@@ -7,42 +7,50 @@ use IPC::Run ();
...
+    my $display = ":$port";
+    my $pid     = fork();
+    die unless defined $pid;
+    if (!$pid) {
+        listen($s, 1);
+        my $peer;
+        accept($peer, $s);
+        close($s);
+        open(STDIN,  "<&", $peer) || die "can't dup client to stdin";
+        open(STDOUT, ">&", $peer) || die "can't dup client to stdout";
```
Comment (okurz) [inline consoles/localXvnc.pm]: the "open" are handled by autodie, same as exec in line 36

**Comment by Soulofdestiny** [comment]
Travis checks failed because ssh.pm was deleted - delete it also from Makefile.am

**Comment by coolo** [comment]
cool that we have test cases, nah? :)

File: consoles/localXvnc.pm (hunk 2 above)
Comment (coolo) [inline consoles/localXvnc.pm]: I copied tutorial code ;)

File: consoles/localXvnc.pm [hunk 3]
```diff
@@ -7,42 +7,50 @@ use IPC::Run ();
...
-    my $display_id = get_var("VNC") || die "VNC unset in vars.json.";
-    my $display = ":" . $display_id;
-    # FIXME: do the full monty xauth authentication, with a local
-    # XAUTHORITY=./XAuthority file
+    # start Xvnc on a random high port and use that port also as $DISPLAY
+
+    my $tcpproto = getprotobyname('tcp');
+    my $s;
+    socket($s, PF_INET, SOCK_STREAM, $tcpproto) || die "socket: $!\n";
+    bind($s, sockaddr_in(0, INADDR_ANY)) || die "bind: $!\n";
```
Comment (aaannz) [inline consoles/localXvnc.pm]: bind is already covered by autodie too

File: backend/svirt.pm [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package backend::svirt;
+use strict;
+use base ('backend::baseclass');
+
+# this is a fake backend to some extend. We do don't start VMs, but provide ssh access
```
Comment (aaannz) [inline backend/svirt.pm]: typo in comment. Is the meaning 'we do not start ...' right?

File: consoles/localXvnc.pm (hunk 3 above)
Comment (coolo) [inline consoles/localXvnc.pm]: and socket is not?

File: backend/svirt.pm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+sub do_savevm {
+    my ($self, $args) = @_;
+    return {};
+}
+
+sub do_loadvm {
+    my ($self, $args) = @_;
+    die "if you need loadvm, you're screwed with IPMI";
```
Comment (aaannz) [inline backend/svirt.pm]: Is this copy&paste or is IPMI really related to this?

File: consoles/localXvnc.pm (hunk 3 above)
Comment (aaannz) [inline consoles/localXvnc.pm]: Doesn't seem so ( http://perldoc.perl.org/autodie.html )

File: backend/svirt.pm (hunk 5 above)
Comment (coolo) [inline backend/svirt.pm]: it's still true, no? :)

**Comment by coolo** [comment]
I merged the test code to avoid @Soulofdestiny and @okurz having to base their changes on a moving branch. But I expect fine tunings to be necessary

**Comment by okurz** [comment]
very sweet of you but you don't need to look after us regarding git, we can handle moving targets as long as you keep us informed what to do and where, it's fine :-)

**Comment by coolo** [comment]
I removed the WIP from the summary - just to avoid any further confusions ;)

**Comment by coolo** [comment]
the svirt backend is still not feature complete, but the wait_serial refactoring is important for s390 too

**Pull request merged**